src/providers/cryptoauthlib/mod.rs

```diff
@@ -273,103 +316,105 @@ impl ProviderBuilder {
+        self.slave_address = slave_address;
 
         self
     }
 
     /// Specify i2c bus for ATECC device
-    pub fn with_bus(mut self, bus: u8) -> ProviderBuilder {
-        self.bus = Some(bus);
+    pub fn with_bus(mut self, bus: Option<u8>) -> ProviderBuilder {
+        self.bus = bus;
 
         self
     }
 
     /// Specify i2c baudrate
-    pub fn with_baud(mut self, baud: u32) -> ProviderBuilder {
-        self.baud = Some(baud);
+    pub fn with_baud(mut self, baud: Option<u32>) -> ProviderBuilder {
+        self.baud = baud;
 
         self
     }
 
     /// Attempt to build CryptoAuthLib Provider
     pub fn build(self) -> std::io::Result<Provider> {
-        let iface_type = match self.iface_type {
+        let iface_cfg = match self.iface_type {
             Some(x) => match x.as_str() {
-                "i2c" => rust_cryptoauthlib::atca_iface_setup_i2c(
-                    self.device_type.ok_or_else(|| {
-                        Error::new(ErrorKind::InvalidData, "missing atecc device type")
-                    })?,
-                    self.wake_delay.ok_or_else(|| {
-                        Error::new(ErrorKind::InvalidData, "missing atecc wake delay")
-                    })?,
-                    self.rx_retries.ok_or_else(|| {
-                        Error::new(
-                            ErrorKind::InvalidData,
-                            "missing rx retries number for atecc",
+                "i2c" => {
+                    let atcai2c_iface_cfg = rust_cryptoauthlib::AtcaIfaceI2c::default()
+                        .set_slave_address(self.slave_address.ok_or_else(|| {
+                            Error::new(ErrorKind::InvalidData, "missing atecc i2c slave address")
+                        })?)
+                        .set_bus(self.bus.ok_or_else(|| {
+                            Error::new(ErrorKind::InvalidData, "missing atecc i2c bus")
+                        })?)
+                        .set_baud(self.baud.ok_or_else(|| {
+                            Error::new(ErrorKind::InvalidData, "missing atecc i2c baud rate")
+                        })?);
+                    rust_cryptoauthlib::AtcaIfaceCfg::default()
+                        .set_iface_type("i2c".to_owned())
+                        .set_devtype(self.device_type.ok_or_else(|| {
+                            Error::new(ErrorKind::InvalidData, "missing atecc device type")
+                        })?)
+                        .set_wake_delay(self.wake_delay.ok_or_else(|| {
+                            Error::new(ErrorKind::InvalidData, "missing atecc wake delay")
+                        })?)
+                        .set_rx_retries(self.rx_retries.ok_or_else(|| {
+                            Error::new(
+                                ErrorKind::InvalidData,
+                                "missing rx retries number for atecc",
+                            )
+                        })?)
```

Comment:
No default value for any of these? ☹️

Reply:
No good defaults. We may think about adding "probing" i.e. iterate through known, usual configurations. I would rather avoid it at the moment.